### PR TITLE
Guess content

### PR DESCRIPTION
**Pull request by giampaolo**

### CODE REVIEW

File: fulltext/__init__.py [hunk 1]
```diff
@@ -236,19 +241,67 @@ def backend_from_mime(mime):
...
+    if magic is None:
+        warn("magic lib is not installed; assuming mime type %r" % (
+            DEFAULT_MIME))
+        return backend_from_mime(DEFAULT_MIME)
+    else:
+        offset = f.tell()
+        try:
+            f.seek(0)
+            chunk = f.read(MAGIC_BUFFER_SIZE)
+            mime = magic.from_buffer(chunk, mime=True)
```
Comment (btimby) [inline fulltext/__init__.py]: `magic` has a `from_file()` function...

File: fulltext/util.py [hunk 2]
```diff
@@ -108,3 +113,39 @@ def run(*cmd, **kwargs):
...
+                      magic_file=pathjoin(BASE_PATH, 'bin', 'magic'))
+    except Exception:
+        warn('Magic is unavailable, type detection degraded')
+        # If all else fails...
+        magic = None
+
+else:
+    # On linux things are simpler. Linter disabled for next line since we
+    # import here for export.
+    import magic  # NOQA
```
Comment (btimby) [inline fulltext/util.py]: For all of this to work, there are some binary files that have to be packaged in `setup.py` and `MANIFEST.in` I am OK with you omitting this for now. It took me a while to get all of this working in duster.

File: fulltext/__init__.py (hunk 1 above)
Comment (giampaolo) [reply fulltext/__init__.py]: In here we're dealing with a file object though

Comment (btimby) [reply fulltext/__init__.py]: You're right, my bad.

**Pull request merged**